osgar/drivers/logzeromq.py

```diff
@@ -0,0 +1,57 @@
+"""
+  Logger for ZeroMQ communication
+"""
+
+from threading import Thread
+
+import zmq
+
+from osgar.bus import BusShutdownException
+
+
+class LogZeroMQ:
+    def __init__(self, config, bus):
+        mode = config['mode']
+        endpoint = config['endpoint']
+
+        context = zmq.Context()
+        print("Connecting to hello world server ...")
+        if mode == 'PULL':
+            self.socket = context.socket(zmq.PULL)
+            self.thread = Thread(target=self.run_input, daemon=True)
+        elif mode == 'PUSH':
+            self.socket = context.socket(zmq.PUSH)
+            self.thread = Thread(target=self.run_output, daemon=True)
+        else:
+            assert False, mode  # unknown/unsupported mode
+
+        self.socket.connect(endpoint)  # TODO config
+
+        self.bus = bus
+
+    def start(self):
+        self.thread.start()
+
+    def join(self, timeout=None):
+        self.thread.join(timeout=timeout)
+
+    def run_input(self):
+        while self.bus.is_alive():
+            message = self.socket.recv()
+            self.bus.publish('raw', message)
+
+    def slot_raw(self, timestamp, data):
+        self.socket.send(data)
```

Comment:
Can this block forever? If so, can we have timeout here as well?

Reply:
Yes, this can block for ever - I think that there is by default queue with several thousands messages until somebody starts to read it ... again I am not sure what should OSGAR do - the idea is to transport everything, no dropped messages.

Reply:
I see `timeout` has been added to the socket. Can we have a test for this? So that we are sure that either send or recv does not block forever - basically just that `request_stop` works in both cases. Thanks.